front/components/ConnectorPermissionsTree.tsx

```diff
@@ -119,6 +124,10 @@ export function PermissionTreeChildren({
     (r) => search.trim().length === 0 || r.title.includes(search)
   );
 
+  const selectedParents = [
+    ...new Set(Object.values(parentsById).flatMap((c) => [...c])),
+  ];
```

Comment:
selectedParents will be the same for all nodes, so also defined at the same level as useParentResourcesById, and then   can be passed as a prop to PermissionTreeChildren recursively